azurerm/internal/services/eventhub/eventhub_cluster_resource.go

```diff
@@ -0,0 +1,184 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateEventHubName(),
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Dedicated_1",
+				}, false),
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmEventHubClusterCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Eventhub.ClusterClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+	log.Printf("[INFO] preparing arguments for Azure ARM EventHub Cluster creation.")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	cluster := eventhub.Cluster{
+		Location: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+		Sku:      expandEventHubClusterSkuName(d.Get("sku_name").(string)),
+	}
+
+	future, err := client.Put(ctx, resourceGroup, name, cluster)
+	if err != nil {
+		return err
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("creating eventhub cluster: %+v", err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil || *read.ID == "" {
+		return fmt.Errorf("cannot read EventHub Cluster %s (resource group %s) ID", name, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmEventHubClusterRead(d, meta)
+}
+
+func resourceArmEventHubClusterRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Eventhub.ClusterClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ClusterID(d.Id())
+	if err != nil {
+		return err
+	}
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("making Read request on Azure EventHub Cluster %q (resource group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("sku_name", flattenEventHubClusterSkuName(resp.Sku))
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmEventHubClusterDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Eventhub.ClusterClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+	id, err := parse.NamespaceAuthorizationRuleID(d.Id())
```

Comment:
Why here we are parsing this ID as a namespace authorization rule instead of cluster?